include/gatekeeper_net.h

```diff
@@ -40,4 +42,14 @@ int gatekeeper_init_network(struct net_config *net_conf);
 
 void gatekeeper_free_network(void);
 
+/*
+ * Each functional block listens on one or more lcores. The lcore
+ * identifier also serves as a queue identifier.
+ */
+static inline uint16_t
+get_queue_id(void)
+{
+	return (uint16_t)rte_lcore_id();
```

Comment:
I've confused `lcore_id` with lcore index. Could you try to call Gatekeeper skipping some cores to see if the current code will work fine?